librz/core/canalysis.c

```diff
@@ -7102,3 +7102,123 @@ RZ_API st64 rz_core_analysis_calls_count(RZ_NONNULL RzCore *core) {
 	}
 	return cov;
 }
+
+static void show_reg_args(RzCore *core, int nargs, RzStrBuf *sb) {
+	int i;
+	char regname[8];
+	if (nargs < 0) {
+		nargs = 4; // default args if not defined
+	}
+	for (i = 0; i < nargs; i++) {
+		snprintf(regname, sizeof(regname), "A%d", i);
+		ut64 v = rz_reg_getv(core->analysis->reg, regname);
+		if (sb) {
+			rz_strbuf_appendf(sb, "%s0x%08" PFMT64x, i ? ", " : "", v);
+		} else {
+			rz_cons_printf("A%d 0x%08" PFMT64x "\n", i, v);
+		}
+	}
+}
+
+RZ_API char *rz_core_analysis_call_args(RzCore *core, ut64 pcv) {
+	const char *cc = rz_analysis_cc_default(core->analysis);
+	RzAnalysisFuncArg *arg;
+	RzListIter *iter;
+	RzListIter *nextele;
+
+	RzStrBuf *sb = rz_strbuf_new("");
+	RzAnalysisOp *op = rz_core_analysis_op(core, pcv, -1);
+	if (!op) {
+		RZ_LOG_ERROR("Fail to get op at 0x%" PFMT64x "\n", pcv);
+		rz_strbuf_free(sb);
+		return NULL;
+	}
+	bool go_on = true;
+	if (op->type != RZ_ANALYSIS_OP_TYPE_CALL) {
+		show_reg_args(core, -1, sb);
+		go_on = false;
+	}
+	const char *fcn_name = NULL;
+	RzAnalysisFunction *fcn;
+	if (go_on) {
```

Comment:
can you remove these `if (go_on)` ? essentially each time there is `go_on = false`, just replace it with `return rz_strbuf_drain(sb);`.